playbooks/kubevirt.yml

```diff
@@ -0,0 +1,260 @@
+      args:
+        creates: /etc/kubernetes/admin.conf
+
+    - name: untaint master
+      command: kubectl taint nodes --all node-role.kubernetes.io/master-
+      register: untaint_master
+      failed_when: false
+      changed_when: untaint_master.rc == 0
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy network namespace
+      command: kubectl apply -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/{{ network_operator }}/namespace.yaml
+      args:
+        creates: /etc/cni/net.d/multus.d/multus.kubeconfig
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy network crd
+      command: kubectl apply -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/{{ network_operator }}/network-addons-config.crd.yaml
+      args:
+        creates: /etc/cni/net.d/multus.d/multus.kubeconfig
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy network operator
+      command: kubectl apply -f https://github.com/kubevirt/cluster-network-addons-operator/releases/download/{{ network_operator }}/operator.yaml
+      args:
+        creates: /etc/cni/net.d/multus.d/multus.kubeconfig
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy flannel
+      command: kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/{{ flannel_version }}/Documentation/kube-flannel.yml
+      args:
+        creates: /etc/cni/net.d/10-flannel.conflist
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy kubevirt operator
+      command: kubectl apply -f https://github.com/kubevirt/kubevirt/releases/download/{{ kubevirt_version }}/kubevirt-operator.yaml
+      args:
+        creates: /var/lib/kubelet/device-plugins/kubevirt-tun.sock
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: deploy kubevirt cr
+      command: kubectl apply -f https://github.com/kubevirt/kubevirt/releases/download/{{ kubevirt_version }}/kubevirt-cr.yaml
+      args:
+        creates: /var/lib/kubelet/device-plugins/kubevirt-tun.sock
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+
+    - name: install virtctl
+      get_url:
+        url: https://github.com/kubevirt/kubevirt/releases/download/{{ kubevirt_version }}/virtctl-{{ kubevirt_version }}-linux-amd64
+        dest: /usr/bin/virtctl
+        mode: u=rwx,g=rx,o=rx
+
+    - name: sleep for 300 seconds
+      wait_for: timeout=300
+
+    - name: check for bridge-foreman NetworkAttachmentDefinition
+      command: kubectl get net-attach-def bridge-foreman
+      environment:
+        KUBECONFIG: /etc/kubernetes/admin.conf
+      failed_when: false
+      changed_when: false
+      register: bridge_foreman_net_attach
+
+    - name: prepare bridge-foreman NetworkAttachmentDefinition file
```

Comment:
You haven't created network addons operator CR specifying which addons (multus, linuxBridge) you want.

Reply:
This is the relevant step from the getting_started.md of the plugin:

https://github.com/theforeman/foreman_kubevirt/blame/master/docs/getting-started.md#L158

Reply:
```
cat <<EOF | kubectl create -f -
---
apiVersion: networkaddonsoperator.network.kubevirt.io/v1alpha1
kind: NetworkAddonsConfig
metadata:
  name: cluster
spec:
  imagePullPolicy: Always
  kubeMacPool: {}
  multus: {}
```

Reply:
@masayag you cannot create a NetworkAttachmentDefinition before Multus is deployed, as it creates the new CRD. Are you saying what should be added to the playbook?

Reply:
Yes, the playbook is following steps that I documented [here](https://github.com/theforeman/foreman_kubevirt/blob/master/docs/getting-started.md#install-kubernetes-and-kubevirt)

Copied the relevant part:
```
kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml

# Check for latest version
RELEASE=v0.16.0
kubectl apply -f https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt.yaml

kubectl apply -f https://raw.githubusercontent.com/kubevirt/ovs-cni/master/examples/kubernetes-ovs-cni.yml

NET_RELEASE=0.5.0
kubectl apply -f https://raw.githubusercontent.com/kubevirt/cluster-network-addons-operator/master/manifests/cluster-network-addons/${NET_RELEASE}/namespace.yaml
kubectl apply -f https://raw.githubusercontent.com/kubevirt/cluster-network-addons-operator/master/manifests/cluster-network-addons/${NET_RELEASE}/network-addons-config.crd.yaml
kubectl apply -f https://raw.githubusercontent.com/kubevirt/cluster-network-addons-operator/master/manifests/cluster-network-addons/${NET_RELEASE}/operator.yaml

cat <<EOF | kubectl create -f -
---
apiVersion: networkaddonsoperator.network.kubevirt.io/v1alpha1
kind: NetworkAddonsConfig
metadata:
  name: cluster
spec:
  imagePullPolicy: Always
  kubeMacPool: {}
  multus: {}
```

Running them according to the order of appearance above ended in a working kubevirt+multus+mac address pool.

If you have any comments on that, please share.

Reply:
If you add the last part to the playbook, it should be all right.

Reply:
yeah, creating a NetworkAddonsConfig helps

Reply:
@evgeni my steps are updated for using OVS, but I think this should be changed for using linux bridge, as specified on [1].

My devenv still using ovs-cni so I haven't walked through the entire list of steps required, and there might be additional issues as well.
I'll take a snapshot of my env and recreate it earlier next week based on linux-bridge, so I'll have the exact steps needed for it.

[1] https://github.com/kubevirt/cluster-network-addons-operator#linux-bridge